src/D2L.CodeStyle.Analyzers/Async/Generator/AsyncToSyncMethodTransformer.cs

```diff
@@ -185,6 +189,18 @@ private StatementSyntax Transform( StatementSyntax stmt )
 			_ => UnhandledSyntax( stmt )
 		};
 
+	private StatementSyntax Transform( ReturnStatementSyntax returnStmt ) {
+		var expr = returnStmt.Expression;
+		if( m_generatedFunctionReturnsVoid && expr is not null ) {
+			if( expr is InvocationExpressionSyntax || expr is AssignmentExpressionSyntax || expr is IIncrementOrDecrementOperation || expr is DeclarationExpressionSyntax || expr is MemberAccessExpressionSyntax ) {
+				return SyntaxFactory.ParseStatement( $"{Transform(expr).WithTriviaFrom(returnStmt)};\nreturn;" );

```

Comment:
```suggestion
				return SyntaxFactory.ParseStatement( $"{Transform(expr).WithTriviaFrom(returnStmt)};" );

```

Do we need the `return;`? I wouldn't go out of my way to remove redundant things, but I also wouldn't go out of my way to add them.

Reply:
I think this could be `SyntaxFactory.ExpressionStatement( Transform( expr ) )` instead of calling into `ParseStatement` in that case?

ExpressionStatement takes an expression as input and "lifts" it up to be a statement. E.g. `Foo()` is an InvocationExpression, and `Foo();` is an ExpressionStatement contain that invocation (note also the semicolon)

Reply:
I believe we need we keep the return afterwards for cases like this:
```
Task FooAsync() {
	if( condition ) {
		return BarAsync(2);
	}
	return BarAsync(4);
}
```
We don't want to transform this into 
```
Task FooAsync() {
	if( condition ) {
		Bar(2);
	}
	Bar(4);
}
```